src/builder-module.c

```diff
@@ -1634,6 +1638,22 @@ builder_module_build_helper (BuilderModule  *self,
           return FALSE;
         }
     }
+  else if (cargo)
+    {
+      cargo_file = find_file_with_extension (source_subdir, "Cargo.toml");
```

Comment:
Hmm, Cargo.lock only contains dependency definitions (and nothing else), so if the build process needed to read anything else like metadata (name / description / version ...) then it would need Cargo.toml to be present too

Reply:
Never mind, I mistook this as being related to cargo sources whereas this is actually about cargo build.